src/crypto/ecc_curve.rs

```diff
@@ -11,12 +12,13 @@ pub enum ECCCurve {
     BrainpoolP384r1,
     BrainpoolP512r1,
     Secp256k1,
+    Unknown(String),
```

Comment:
I would prefer to store the parsed Oid here and not the string

Reply:
I'd also prefer that, but couldn't think of a way to implement `ECCCurve::oid_str` (which returns a `&str`), then. Maybe the API of `ECCCurve` needs some more adjustment?

Reply:
(I did already incompatibly change that part of the public API anyway, by removing the `Copy` derive, anyway. And changed some functions to take a reference instead of an owned `ECCCurve`)

Reply:
returning String is fine

Reply:
I've adjusted the change to `ECCCurve` accordingly.